koin-projects/koin-androidx-viewmodel/src/main/java/org/koin/androidx/viewmodel/ext/android/ActivityExt.kt

```diff
@@ -29,35 +31,42 @@ import kotlin.reflect.KClass
  *
  * @author Arnaud Giuliani
  */
-
-fun <T : ViewModel> ViewModelStoreOwner.viewModel(
-    clazz: KClass<T>,
+inline fun <reified T : ViewModel> AppCompatActivity.viewModel(
     qualifier: Qualifier? = null,
-    parameters: ParametersDefinition? = null
+    noinline state: BundleDefinition? = null,
+    noinline owner: ViewModelOwnerDefinition = { from(this,this) },
+    noinline parameters: ParametersDefinition? = null
 ): Lazy<T> {
-    return lazy(LazyThreadSafetyMode.NONE) { getViewModel(clazz, qualifier, parameters) }
+    return lazy(LazyThreadSafetyMode.NONE) {
+        getViewModel<T>(qualifier, state, owner, parameters)
+    }
 }
 
-inline fun <reified T : ViewModel> ViewModelStoreOwner.viewModel(
+fun <T : ViewModel> AppCompatActivity.viewModel(
     qualifier: Qualifier? = null,
-    noinline parameters: ParametersDefinition? = null
+    state: BundleDefinition? = null,
+    owner: ViewModelOwnerDefinition = { from(this,this) },
+    clazz: KClass<T>,
+    parameters: ParametersDefinition? = null
 ): Lazy<T> {
-    return lazy(LazyThreadSafetyMode.NONE) { getViewModel(T::class, qualifier, parameters) }
+    return lazy(LazyThreadSafetyMode.NONE) { getViewModel(qualifier, state, owner, clazz, parameters) }
 }
 
-inline fun <reified T : ViewModel> ViewModelStoreOwner.getViewModel(
+inline fun <reified T : ViewModel> AppCompatActivity.getViewModel(
     qualifier: Qualifier? = null,
+    noinline state: BundleDefinition? = null,
+    noinline owner: ViewModelOwnerDefinition = { from(this,this) },
     noinline parameters: ParametersDefinition? = null
 ): T {
-    return getViewModel(T::class, qualifier, parameters)
+    return getViewModel(qualifier, state, owner, T::class, parameters)
 }
 
-fun <T : ViewModel> ViewModelStoreOwner.getViewModel(
-    clazz: KClass<T>,
+fun <T : ViewModel> AppCompatActivity.getViewModel(
```

Comment:
Better to use [ComponentActivity](https://developer.android.com/reference/androidx/activity/ComponentActivity)